oddt/fingerprints.py

```diff
@@ -0,0 +1,173 @@
+
+"""
+Module checks interactions between two molecules.
+"""
+
+import numpy as np
+import oddt
+from oddt.interactions import close_contacts, pi_stacking, pi_cation, \
+    hbond_acceptor_donor, salt_bridge_plus_minus, hydrophobic_contacts, acceptor_metal
+
+__all__ = ['InteractionFingerprint', 'SimpleInteractionFingerprint']
+
+
+def InteractionFingerprint(protein, ligand, strict=None):
+    """
+        Interaction fingerprint accomplished by converting the molecular
+        interaction of ligand-protein into bit array according to
+        the residue of choice and the interaction. For every residue
+        (One row = one residue) there are eight bits which represent
+        seven type of interactions:
+        (Column 0) hydrophobic contacts
+        (Column 1) aromatic face to face
+        (Column 2) aromatic edge to face
+        (Column 3) hydrogen bond (protein as hydrogen bond donor)
+        (Column 4) hydrogen bond (protein as hydrogen bond acceptor)
+        (Column 5) salt bridges (protein positively charged)
+        (Column 6) salt bridges (protein negatively charged)
+        (Column 7) salt bridges (ionic bond with metal ion)
+
+        Parameters
+        ----------
+        protein, ligand : oddt.toolkit.Molecule object
+            Molecules to check interactions beetwen them
+        strict : bool, (deafult = None)
+            If 'True', include condition, which informs whether atoms
+            form 'strict' H-bond (pass all angular cutoffs).
+        Returns
+        -------
+        InteractionFingerptint : numpy array
+            Vector of calculated IFP
+            Size = (rows = 1, columns = no residues * 8 type of interaction)
+
+    """
+    resids = np.unique(protein.atom_dict['resid'])
+    IFP = np.zeros((len(resids), 8), dtype=np.uint8)
+
+    # hydrophobic contacts (column = 0)
```

Comment:
maybe you can find all types of contacts in a loop? (disclaimer: I'm not sure it's a good idea)